src/pc/1.13/chunk.js

```diff
@@ -0,0 +1,454 @@
+      }
+    ]
+  },
+  {
+    'name': 'dataArray',
+    'type': ['buffer', {
+      'countType': ['longToByte', {
+        'type': 'varint'
+      }]
+    }]
+  },
+  {
+    'name': 'blockLight',
+    'type': ['buffer', {
+      'count': 16 * 16 * 16 / 2
+    }]
+  }
+  ]]
+
+  const indirectPalette = ['array', {
+    'type': 'varint',
+    'countType': 'varint'
+  }]
+
+  Chunk.packingProtocol = new ProtoDef()
+  Chunk.packingProtocol.addType('indirectPalette', indirectPalette)
+  Chunk.packingProtocol.addType('longToByte', longToByte)
+  Chunk.packingProtocol.addType('section', p)
+  Chunk.packingProtocol.addType('sectionNoSkylight', pns)
+
+  Chunk.w = w
+  Chunk.l = l
+  Chunk.h = h
+  Chunk.BUFFER_SIZE = BUFFER_SIZE
+  return Chunk
+}
+
+var Block
+
+const exists = val => val !== undefined
+const getArrayPosition = (pos) => pos.x + w * (pos.z + l * pos.y)
+const getBlockCursor = (pos) => getArrayPosition(pos) << 1
+const getBlockLightCursor = (pos) => (getArrayPosition(pos) >>> 1) + (CHUNK_VOLUME << 1)
+const getSkyLightCursor = (pos) => (getArrayPosition(pos) >>> 1) + (CHUNK_VOLUME >>> 1) * 5
+const getBiomeCursor = (pos) => X3_CHUNK_VOLUME + (pos.z * w) + pos.x
+
+class Chunk {
+  constructor () {
+    this.data = Buffer.alloc(BUFFER_SIZE)
+  }
+
+  initialize (iniFunc) {
+    const skylight = (CHUNK_VOLUME >>> 1) * 5
+    const light = CHUNK_VOLUME << 1
+
+    let biome = X3_CHUNK_VOLUME - 1
+
+    const data = this.data
+    let y
+    for (let n = 0; n < CHUNK_VOLUME; n++) {
+      // x = n & 15;
+      // z = (n & 255) >>> 4;
+      y = n >>> 8
+      const block = iniFunc(n & 15, y, (n & 255) >>> 4, n)
+      if (block !== null) {
+        data.writeUInt16LE(block.type << 4 | block.metadata, n << 1)
+        writeUInt4LE(data, block.light, (n >>> 3) + light)
+        writeUInt4LE(data, block.skyLight, (n >>> 3) + skylight)
+        if (y === 0) {
+          biome++
+          data.writeUInt8(block.biome.id || 0, biome)
+        }
+      } else if (y === 0) {
+        biome++
+      }
+    }
+  }
+
+  getBlock (pos) {
+    var block = new Block(this.getBlockType(pos), this.getBiome(pos), this.getBlockData(pos))
+    block.light = this.getBlockLight(pos)
+    block.skyLight = this.getSkyLight(pos)
+    return block
+  }
+
+  setBlock (pos, block) {
+    if (exists(block.type)) { this.setBlockType(pos, block.type) }
+    if (exists(block.metadata)) { this.setBlockData(pos, block.metadata) }
+    if (exists(block.biome)) { this.setBiome(pos, block.biome.id) }
+    if (exists(block.skyLight)) { this.setSkyLight(pos, block.skyLight) }
+    if (exists(block.light)) { this.setBlockLight(pos, block.light) }
+  }
+
+  getBiomeColor (pos) {
+    // polyfill
+    return {
+      r: 0,
+      g: 0,
+      b: 0
+    }
+  }
+
+  setBiomeColor (pos, r, g, b) {
+    // polyfill
+  }
+
+  getBlockType (pos) {
+    var cursor = getBlockCursor(pos)
+    let a = this.data.readUInt16LE(cursor)
+    if (a === 8575) a = 383 // Some fix with the chunk 23..
```

Comment:
Why is this needed ?
It seems unlikely this is the only case where it would happen.

Reply:
I am still working on something about that.